AndroidSDKCore/src/main/java/com/leanplum/internal/Request.java

```diff
@@ -480,22 +480,83 @@ protected Void doInBackground(Void... params) {
     });
   }
 
-  private void sendRequests() {
+
+  static class RequestsWithEncoding {
+    List<Map<String, Object>> unsentRequests;
+    List<Map<String, Object>> requestsToSend;
+    String jsonEncodedString;
+  }
+
+  private RequestsWithEncoding getRequestsWithEncodedStringForErrors() {
+
     List<Map<String, Object>> unsentRequests = new ArrayList<>();
     List<Map<String, Object>> requestsToSend;
+    String jsonEncodedRequestsToSend;
+
+
+    String uuid = UUID.randomUUID().toString();
+    for (Map<String, Object> error : localErrors) {
+      error.put(UUID_KEY, uuid);
+      unsentRequests.add(error);
+    }
+    requestsToSend = unsentRequests;
+    jsonEncodedRequestsToSend = jsonEncodeUnsentRequests(unsentRequests);
+
+    RequestsWithEncoding requestsWithEncoding = new RequestsWithEncoding();
+    requestsWithEncoding.unsentRequests = unsentRequests;
+    requestsWithEncoding.requestsToSend = requestsToSend;
+    requestsWithEncoding.jsonEncodedString = jsonEncodedRequestsToSend;
+
+    return requestsWithEncoding;
+  }
+
+
+  protected RequestsWithEncoding getRequestsWithEncodedStringStoredRequests(double fraction) {
+    try {
+      List<Map<String, Object>> unsentRequests;
+      List<Map<String, Object>> requestsToSend;
+      String jsonEncodedRequestsToSend;
+      RequestsWithEncoding requestsWithEncoding = new RequestsWithEncoding();
+
+      if (fraction < 0.01) {
+        unsentRequests = new ArrayList<>(0);
+        requestsToSend = new ArrayList<>(0);
+      } else {
+        unsentRequests = getUnsentRequests(fraction);
+        requestsToSend = removeIrrelevantBackgroundStartRequests(unsentRequests);
+      }
+
+      jsonEncodedRequestsToSend = jsonEncodeUnsentRequests(unsentRequests);
+
+      requestsWithEncoding.unsentRequests = unsentRequests;
+      requestsWithEncoding.requestsToSend = requestsToSend;
+      requestsWithEncoding.jsonEncodedString = jsonEncodedRequestsToSend;
+
+      return requestsWithEncoding;
+    } catch (OutOfMemoryError E) {
```

Comment:
Maybe a comment within the catch block suggesting why we are catching OOO and still returning requests with 50% multiplier